src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData-CSharp.Web/Services/Ingestion/PdfPigReader.cs

```diff
@@ -0,0 +1,41 @@
+using Microsoft.Extensions.DataIngestion;
+using Microsoft.SemanticKernel.Text;
+using UglyToad.PdfPig;
+using UglyToad.PdfPig.Content;
+using UglyToad.PdfPig.DocumentLayoutAnalysis.PageSegmenter;
+using UglyToad.PdfPig.DocumentLayoutAnalysis.WordExtractor;
+
+namespace ChatWithCustomData_CSharp.Web.Services.Ingestion;
+
+internal sealed class PdfPigReader : IngestionDocumentReader
+{
+    public override Task<IngestionDocument> ReadAsync(Stream source, string identifier, string mediaType, CancellationToken cancellationToken = default)
+    {
+        using var pdf = PdfDocument.Open(source);
+        var document = new IngestionDocument(identifier);
+        foreach (var section in pdf.GetPages().SelectMany(GetPageSections))
+        {
+            document.Sections.Add(section);
+        }
```

Comment:
Single page should map to a single section.

```suggestion
        foreach (var page in pdf.GetPages())
        {
            document.Sections.Add(GetPageSection(page));
        }
```